### PR TITLE
[exporter/syslog]: fix setting network connection

### PR DESCRIPTION
**Pull request by kasia-kujawa**
**Description:**  fix setting network connection, do not load TLS configuration for UDP

**Link to tracking Issue:** #31130 

**Testing:** unit test, manual tests with syslog server

**Documentation:**  added information that TLS config is applied only when TCP is used

### CODE REVIEW

**Review by andrzej-stencel — APPROVED** [review]
Tested with both UDP and TCP, works like a charm :ok_hand: Thank you!

File: exporter/syslogexporter/sender.go [hunk 1]
```diff
@@ -78,8 +78,8 @@ func (s *sender) dial() error {
 		s.conn = nil
 	}
 	var err error
-	if s.tlsConfig != nil {
-		s.conn, err = tls.Dial("tcp", s.addr, s.tlsConfig)
+	if s.tlsConfig != nil && s.network == "tcp" {
```
Comment (rnishtala-sumo) [inline exporter/syslogexporter/sender.go:81]: nit: maybe a constant for "tcp" would be ideal

**Review by rnishtala-sumo — APPROVED** [review]
LGTM! minor comment on tcp/udp strings

File: exporter/syslogexporter/sender.go (hunk 1 above)
Comment (kkujawa-sumo) [reply exporter/syslogexporter/sender.go:81]: I'll open a new pull request to change tcp/udp strings into constants, we use these strings in several places.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**